ns-convert

```diff
@@ -1,184 +1,341 @@
-                                                   'version=',
-                                                   ])
-    output = None
-    for opt, arg in opts:
-        if opt in ("-o", "--output"):
-            output = arg
-
-    if len(rem) != 1:
-        print("Wrong number of arguments")
-        return -1
-
-    filename = rem[0]
-    ci = ConsoleIndicator()
-    converter = Converter(filename, output, progress=ci)
-    converter.convert()
-    ci.cleanup()
-    return 0
-
-
-if __name__ == "__main__":
-    res = main()
-    sys.exit(res)
+#!/usr/bin/env python
+from __future__ import print_function
+
+import os
+import sys
+import neuroshare as ns
+import numpy as np
+import time
+import nix
+import h5py
+import argparse
+
+class ProgressIndicator(object):
+    def __init__(self, offset=0):
+        self._cur_value = offset
+        self._max_value = 0
+
+    def setup(self, max_value):
+        self._max_value = max_value
+        self.progress(self._max_value, 0)
+
+    def __add__(self, other):
+        self._cur_value += other
+        self.progress(self._max_value, self._cur_value)
+        return self
+
+    def progress(self, max_value, cur_value):
+        pass
+
+#CLASS DEFINITION FOR NIX(Default) CONVERSION
+class Converter(object):
+    def __init__(self, filepath, output=None, progress=None):
+        if not output:
+            (basefile, ext) = os.path.splitext(filepath)
+            output = "%s.h5" % basefile
+
+        nf = ns.File(filepath)
+        nixF = nix.File.open(output,nix.FileMode.Overwrite)
+        nixB = nixF.create_block("main_block","nix.session")
+        nixS = nixF.create_section("main_section","odml.recording")
+         
+        self._nf = nf
+        self._nixF = nixF
+        self._nixB = nixB
+        self._nixS = nixS
+        self._groups = {}
+        self._sections = {}
+        self.convert_map = {1: self.convert_event,
+                            2: self.convert_analog,
+                            3: self.convert_segment,
+                            4: self.convert_neural}
+        if not progress:
+            progress = ProgressIndicator()
+        self._progress = progress
+
+    def get_group_for_type(self, entity_type):
+        name_map = {1: 'Event',
+                    2: 'Analog',
+                    3: 'Segment',
+                    4: 'Neural'}
+
+        if entity_type not in self._groups:
+            name = name_map[entity_type]
+            group = self._nixB.create_group(name,"group")
+            self._groups[entity_type] = group
+
+        return self._groups[entity_type]
+        
+    def get_section_for_type(self, entity_type):
+        name_map = {1: 'events',
+                    2: 'analogs',
+                    3: 'segments',
+                    4: 'neurals'}
+
+        if entity_type not in self._sections:
+            name    = name_map[entity_type]
+            section = self._nixS.create_section(name,"odml.recording")

```

Comment:
I like the `=` aligned too, but that will give a lot of PEP-8 warnings in a modern IDE, so let's rather not align them.